fhir/com.b2international.snowowl.fhir.rest/src/com/b2international/snowowl/fhir/rest/MetadataController.java

```diff
@@ -0,0 +1,387 @@
+import org.springdoc.webmvc.core.RouterFunctionProvider;
+import org.springframework.beans.factory.ObjectFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.servlet.mvc.method.RequestMappingInfoHandlerMapping;
+
+import com.b2international.commons.exceptions.NotFoundException;
+import com.b2international.snowowl.core.events.util.Promise;
+import static com.b2international.snowowl.core.rest.OpenAPIExtensions.*;
+import com.b2international.snowowl.fhir.core.codesystems.CapabilityStatementKind;
+import com.b2international.snowowl.fhir.core.codesystems.OperationKind;
+import com.b2international.snowowl.fhir.core.codesystems.PublicationStatus;
+import com.b2international.snowowl.fhir.core.codesystems.RestfulCapabilityMode;
+import com.b2international.snowowl.fhir.core.model.StringExtension;
+import com.b2international.snowowl.fhir.core.model.capabilitystatement.*;
+import com.b2international.snowowl.fhir.core.model.capabilitystatement.Rest.Builder;
+import com.b2international.snowowl.fhir.core.model.dt.Code;
+import com.b2international.snowowl.fhir.core.model.dt.Uri;
+import com.b2international.snowowl.fhir.core.model.operationdefinition.OperationDefinition;
+import com.b2international.snowowl.fhir.core.model.operationdefinition.Parameter;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.Maps;
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.extensions.Extension;
+import io.swagger.v3.oas.annotations.extensions.ExtensionProperty;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import io.swagger.v3.oas.annotations.responses.ApiResponses;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import io.swagger.v3.oas.models.OpenAPI;
+import io.swagger.v3.oas.models.PathItem;
+import io.swagger.v3.oas.models.Paths;
+import io.swagger.v3.oas.models.media.Schema;
+
+/**
+ * REST end-point for serving content describing the server's capabilities.
+ * @since 8.0.0
+ */
+@Tag(description="CapabilityStatement", name = "CapabilityStatement", extensions = 
+	@Extension(name = B2I_OPENAPI_X_NAME, properties = { 
+		  @ExtensionProperty(name = B2I_OPENAPI_PROFILE, value = "http://hl7.org/fhir/StructureDefinition/CapabilityStatement")
+	}))
+@RestController
+@RequestMapping(value="/", produces = { AbstractFhirResourceController.APPLICATION_FHIR_JSON })
+public class MetadataController extends AbstractFhirResourceController<CapabilityStatement> {
+	
+	/*
+	 * Class to expose the getOpenApi() method on the superclass.
+	 */
+	@Component
+	public static class SnowOwlOpenApiWebMvcResource extends OpenApiWebMvcResource {
```

Comment:
Due to possible duplicate reading and caching of API definition, I suggest moving to the core.rest module instead of declaring it in the fhir.rest wrapper.